### PR TITLE
Add trimTrailingWhitespace Editor Option

### PR DESCRIPTION
**Pull request by jessetrinity**
[EditorConfig.org](https://editorconfig.org/) describes `trimTrailingWhitespace` as such:

> trim_trailing_whitespace: set to true to remove any whitespace characters preceding newline characters and false to ensure it doesn't.

Currently tsserver always performs formatting operations as though this setting were `true`. This change simply allows `.editorconfig` to setting `trim_triling_whitespace = false` to preserve whitespace so that the trailing whitespace denoted by the cursors below is not removed.

```ts

var a;     |<
     |<
while (true){     |<
     |<
}      |<
```
This change _should not_ change the default formatting behavior. Having `trim_trailing_whitespace = true` or the absence of the `.editorconfig` setting should exhibit the existing behavior.

fourslash tests based on [formatDocumentWithTrivia](https://github.com/microsoft/TypeScript/blob/master/tests/cases/fourslash/formatDocumentWithTrivia.ts) and [formatSelectionWithTrivia](https://github.com/microsoft/TypeScript/blob/master/tests/cases/fourslash/formatSelectionWithTrivia.ts)

### CODE REVIEW

**Comment by jessetrinity** [comment]
bad bot

**Comment by jessetrinity** [comment]
@DanielRosenwasser @mjbvz FYI

**Comment by DanielRosenwasser** [comment]
The bot is good and keeps us humble, reminding us just how little our status matters to it.

File: src/services/formatting/formatting.ts [hunk 1]
```diff
@@ -427,7 +427,9 @@ namespace ts.formatting {
             if (leadingTrivia) {
                 indentTriviaItems(leadingTrivia, initialIndentation, /*indentNextTokenOrTrivia*/ false,
                     item => processRange(item, sourceFile.getLineAndCharacterOfPosition(item.pos), enclosingNode, enclosingNode, /*dynamicIndentation*/ undefined!));
-                trimTrailingWhitespacesForRemainingRange();
+                if(options.trimTrailingWhitespace !== false){
```
Comment (andrewbranch) [inline src/services/formatting/formatting.ts]:
```suggestion
                if (options.trimTrailingWhitespace !== false) {
```

Comment (jessetrinity) [reply src/services/formatting/formatting.ts]: A whitespace formatting error on a PR about whitespace formatting? 🤔

**Comment by sandersn** [comment]
@jessetrinity this looks ready; are we holding it until something else is done?

**Pull request merged**